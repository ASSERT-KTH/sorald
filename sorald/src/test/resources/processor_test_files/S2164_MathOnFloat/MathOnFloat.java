// Test for rule s2164

class MathOnFloat {

    float e = 2.71f;
    float pi = 3.14f;
    double c = e * pi; // Noncompliant

    // Tests from https://github.com/SonarSource/sonar-java/blob/master/java-checks-test-sources/src/main/java/checks/MathOnFloatCheck.java
    void myMethod() {
        float a = 16777216.0f;
        float b = 1.0f;

        double d1 = a + b; // Noncompliant ; addition is still between 2 floats
        double d2 = a - b; // Noncompliant
        double d3 = a * b; // Noncompliant
        double d4 = a / b; // Noncompliant
        double d5 = a / b + b; // Noncompliant

        double d6 = a + d1;

        double d7 = a + a / b; // Noncompliant

        int i = 16777216;
        int j = 1;
        int k = i + j;
        System.out.println("[Max time to retrieve connection:"+(a/1000f/1000f)+" ms.");
        System.out.println("[Max time to retrieve connection:"+a/1000f/1000f);
    }

}
