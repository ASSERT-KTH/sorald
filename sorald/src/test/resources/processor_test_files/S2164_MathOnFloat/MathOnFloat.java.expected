
// Test for rule s2164

class MathOnFloat {

    float e = 2.71f;
    float pi = 3.14f;
    double c = (double) e * pi;

    // Tests from https://github.com/SonarSource/sonar-java/blob/master/java-checks-test-sources/src/main/java/checks/MathOnFloatCheck.java
    void myMethod() {
        float a = 16777216.0f;
        float b = 1.0f;

        double d1 = ((double) (a)) + b;
        double d2 = ((double) (a)) - b;
        double d3 = ((double) (a)) * b;
        double d4 = ((double) (a)) / b;
        double d5 = ((((double) (a)) / b)) + b;

        double d6 = a + d1;

        double d7 = a + ((double) (a)) / b;

        double d8 = (((double) (a)) + b) + (((double) (e)) * pi);

        int i = 16777216;
        int j = 1;
        int k = i + j;
        System.out.println("[Max time to retrieve connection:"+(a/1000f/1000f)+" ms.");
        System.out.println("[Max time to retrieve connection:"+a/1000f/1000f);
    }

}
